src/main/java/net/javadiscord/javabot/listener/CodeAttachmentListener.java

```diff
@@ -0,0 +1,123 @@
+package net.javadiscord.javabot.listener;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Listens for messages containing "message.txt" or similar file attachments
+ * containing source code, in order to format the code into one or more code
+ * snippet blocks in the Discord channel instead. The code blocks are sent in
+ * a series of messages that chain their replies to the original message.
+ */
+public class CodeAttachmentListener extends ListenerAdapter {
+	@Override
+	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		if (isCodeMessage(event.getMessage())) {
+			fetchSource(event.getMessage())
+				.thenApply(this::processSource)
+				.thenAcceptAsync(blocks -> {
+					Message lastMsg = event.getMessage();
+					for (var block : blocks) {
+						lastMsg = lastMsg.reply(block).complete();
```

Comment:
Aside from that, why did you use a blocking `complete()` call here?

Reply:
This way, I am able to chain the message replies such that the first code block replies to the original message, then the next code block replies to the first code block, and then the third code block replies to the second code block, and so on. And I do this using an async block that will be run in whatever executor is being used.

Reply:
Note that this would still be possible using `.submit()` instead of `complete()` but I guess it's fine as it happens asynchronously anyways.